main.nf

```diff
@@ -12,43 +12,50 @@
 nextflow.enable.dsl = 2
 
 /*
-========================================================================================
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     GENOME PARAMETER VALUES
-========================================================================================
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-params.fasta   = WorkflowMain.getGenomeAttribute(params, 'fasta')
-params.gtf     = WorkflowMain.getGenomeAttribute(params, 'gtf')
-params.aligner = params.release == 1 ? 'hisat2' : 'bowtie2'
+params.fasta         = WorkflowMain.getGenomeAttribute(params, 'fasta')
+params.gtf           = WorkflowMain.getGenomeAttribute(params, 'gtf')
+params.bowtie2_index = WorkflowMain.getGenomeAttribute(params, 'bowtie2')
+params.star_index    = WorkflowMain.getGenomeAttribute(params, 'star')
 
 /*
-========================================================================================
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     VALIDATE & PRINT PARAMETER SUMMARY
-========================================================================================
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
 WorkflowMain.initialise(workflow, params, log)
 
 /*
-========================================================================================
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     NAMED WORKFLOW FOR PIPELINE
-========================================================================================
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
+include { BUILD_REFERENCES } from './workflows/build_references'
+include { MARSSEQ          } from './workflows/marsseq'
+
 //
 // WORKFLOW: Run main nf-core/marsseq analysis pipeline
 //
 workflow NFCORE_MARSSEQ {
 
-    include { MARSSEQ } from './workflows/marsseq'
-    MARSSEQ ()
+    if (params.build_references) {
+        BUILD_REFERENCES ()
+    } else {
+        MARSSEQ ()
```

Comment:
no possibility to build references and then run the pipeline in one go?

Reply:
I've made an issue for future: https://github.com/nf-core/marsseq/issues/2